### ISSUE TITLE
Suggestion: Disable GPIO 14 pull up on init

### ISSUE THREAD

**Issue by Fonger** [maximkulkin/esp-ir#9]
# TL;DR

I find that esp8266 GPIO14(D5) although it's not high on boot/after boot, it is "pull-up" by default on boot and even after boot. So I suggest disable pull-up at user_init() to prevent IR led  from burning.

```c
void user_init(void) {
    uart_set_baud(0, 115200);
    gpio_set_pullup(14, false, false); // or gpio_write(14, false);
    //....
    //....
    ir_tx_init(); // or put inside this function, but I call this after homekit is paried. Maybe too late.
}
```

# Experiment

You may do some experiments yourself. I want to add a technical reference to prove my result but I can't find it because most people care only about HIGH/LOW during boot state. You can find conclusion elsewhere: only D1(GPIO4) D2(GPIO5) are the safe pins to use. They are always low. However, i2s WS pin can't use these pins.

Just replace your IR led with visible light led and proper resistor. You can find the LED light "slightly" even after boot. This is because GPIO 14 is pulled up by default and has small voltage. This doesn't matter if you connect IR led to GPIO14/GND because the current is far too small.
![P_20190917_091247](https://user-images.githubusercontent.com/5862369/65003531-7b7b2580-d92b-11e9-9c2b-2c73fec492d4.jpg)

However, If GPIO14 this is connected to NPN transistor base to switch on/off the IR led, then **the IR led will be ON state whole time until you send any IR signal at least one time! (because clr_carrier() explicitly write that pin to low).** I use 1k ohms resistor for my 2N2222 base and this GPIO 14 pull-up voltage is still high enough to switch on the IR led. (I replace IR led with Red LED for demostration)

![P_20190917_083007](https://user-images.githubusercontent.com/5862369/65002347-f8a39c00-d925-11e9-9095-d3fbd72110ad.jpg)

## My Story behind this suggestion

I use very low resistor (10 ohm, not 220 ohms for RED LED I demonstrated above) for my IR led because my esp8266 board is far away from my AC. It works totally fine because IR led should only turn on for short pulse during signal transmission. And it's 50% duty cycle hence reduce effective current by half.

Yesterday I find I broke my IR LED. I moved my board to another place, hence the power to my NodeMCU board was temperately down and back thus the system reboot. Actually I had done this several times without any issues. The only difference is that: this time I don't test if it still works after boot so it doesn't send at least one IR led signal ( no carrier_clr() write GPIO14 to low). And about one day later, I find it no longer works. I think the reason is that the IR led stay on with high current > 24 hours so IR led burn. I want to share this result and suggestion. Thanks for reading this far.

Tested and reproducible on hardware:
- NodeMCU 1.0, ( with CP2102 Serial Chip) (3 boards)

Tested and reproducible on firmware:
- esp-open-rtos
- ESP8266_RTOS_SDK v3.2

GPIO 14 is not HIGH, but pull up with small voltage by default during flash/boot/after boot.

**Comment by maximkulkin** [comment]
It's good to know, but I do not think that this is something that should be handled inside this library (as you have mentioned that it happened prior to library initialization).
Also, this library's initialization is pretty lightweight, so you can do it on startup, no need to postpone it after homekit is paired.

**Comment by peros550** [comment]
Thank you for the suggestion @Fonger !
I was doing many tests (trial and error)  with resistors , transistors in order to find the best combination that would provide reliable IR transmission. 

Sometimes the transistor run really hot right after a reboot. I think I may have burned one transistor.

Now I know why and included the function to my hardware init.   Perhaps it's not bad idea to include the above command in the initialization function.

**Comment by Fonger** [comment]
@peros550 Yeah. Or at least show this caveat in README
